src/main/java/game/Spawner.java

```diff
@@ -49,20 +51,50 @@ public Spawner(final Game game) {
 	 * This method is called every tick.
 	 */
 	public void update() {
+		if (thisGame.getGamestate().isBoss() && thisGame.enemies() < 1 
+				&& System.currentTimeMillis() - startRest > REST) {
+			spawnBoss();
+		} else {
+			updateSaucer();
+			updatePowerup();
+			if (thisGame.enemies() != 0) {
+				startRest = System.currentTimeMillis();
+			}
+			updateWave();
+		}
+	}
+	
+	/**
+	 * Checks if a saucer should be added and does so if needed.
+	 */
+	private void updateSaucer() {
 		if (System.currentTimeMillis() - startSaucerTime > SAUCER_TIME) {
 			spawnSaucer();
 			Logger.getInstance().log("Saucer was spawned");
 			startSaucerTime = System.currentTimeMillis();
 		}
+	}
+	
+	/**
+	 * Checks if a powerup should be added and does so if needed.
+	 */
+	private void updatePowerup() {
 		if (System.currentTimeMillis() - startPowerupTime > POWERUP_TIME) {
 			spawnPowerup();
 			Logger.getInstance().log("Powerup was spawned");
 			startPowerupTime = System.currentTimeMillis();
 		}
-		if (thisGame.enemies() != 0) {
+	}
+	
+	/**
+	 * Checks if the wave should be updated and does so if needed.
+	 */
+	private void updateWave() {
+		if ((startRest == 0 || System.currentTimeMillis() - startRest > REST) && wave % WAVES_BETWEEN_BOSSES == 0) {
+			spawnBoss();
```

Comment:
Boss mode is meer een training mode/manier om makkelijk de boss te zien zonder dat je door een heel aantal levels moet spelen. Geen boss op wave 0 is een goed punt though.